applications/luci-app-transmission/luasrc/model/cbi/transmission.lua

```diff
@@ -1,221 +0,0 @@
--- Copyright 2012 Gabor Varga <vargagab@gmail.com>
--- Licensed to the public under the Apache License 2.0.
-
-require("luci.sys")
-require("luci.util")
-require("luci.model.ipkg")
-
-local uci = require "luci.model.uci".cursor()
-local trport = uci:get_first("transmission", "transmission", "rpc_port") or 9091
-local running = (luci.sys.call("pidof transmission-daemon > /dev/null") == 0)
-local webinstalled = luci.model.ipkg.installed("transmission-web") or uci:get_first("transmission", "transmission", "web_home")
-local button = ""
-if running and webinstalled then
-	button = "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<input type=\"button\" value=\" " .. translate("Open Web Interface") .. " \" onclick=\"window.open('http://'+window.location.hostname+':" .. trport .. "')\"/>"
-end
```

Comment:
Should I implement this feature?

Reply:
I could imagine that it is useful... maybe even as iframe in a modal?

Reply:
However there's no `luci.model.ipkg` for client-side. Can I use luci-app-opkg opkg.js?

Reply:
The old `luci.model.ipkg` class was very resource intensive and heavy. I'd simply replace the check with an `fs.stat()` call, testing whether a file provided by the `transmission-web` is installed. A good candidate would be `/usr/share/transmission/web/index.html`:

```
L.resolveDefault(fs.stat('/usr/share/transmission/web/index.html')).then(function(res) {
    if (res)
        alert('Transmission-web appears to be installed');
});
```